jpo-geojsonconverter/src/main/java/us/dot/its/jpo/geojsonconverter/converter/bsm/BsmTopology.java

```diff
@@ -0,0 +1,83 @@
+import us.dot.its.jpo.geojsonconverter.pojos.geojson.bsm.ProcessedBsm;
+import us.dot.its.jpo.geojsonconverter.serialization.JsonSerdes;
+import us.dot.its.jpo.geojsonconverter.validator.BsmJsonValidator;
+import us.dot.its.jpo.geojsonconverter.validator.JsonValidatorResult;
+
+/**
+ * Kafka Streams Topology builder for processing BSM messages from
+ * ODE BSM JSON -> BSM GeoJSON
+ */
+public class BsmTopology {
+
+    private static final Logger logger = LoggerFactory.getLogger(BsmTopology.class);
+
+    public static Topology build(String bsmOdeJsonTopic, String bsmProcessedJsonTopic, BsmJsonValidator bsmJsonValidator) {
+        StreamsBuilder builder = new StreamsBuilder();
+
+        // Stream for raw BSM messages
+        KStream<Void, Bytes> rawOdeBsmStream =
+            builder.stream(
+                bsmOdeJsonTopic,
+                Consumed.with(
+                    Serdes.Void(),  // Raw topic has no key
+                    Serdes.Bytes()  // Raw JSON bytes
+                )
+            );
+
+        // Validate the JSON and write validation errors to the log at warn level
+        // Passes the raw JSON along unchanged, even if there are validation errors.
+        KStream<Void, DeserializedRawBsm> validatedOdeBsmStream = 
+            rawOdeBsmStream.mapValues(
+                (Void key, Bytes value) -> {
+                    DeserializedRawBsm deserializedRawBsm = new DeserializedRawBsm();
+                    try {
+                        JsonValidatorResult validationResults = bsmJsonValidator.validate(value.get());
+                        deserializedRawBsm.setOdeBsmData(JsonSerdes.OdeBsm().deserializer().deserialize(bsmOdeJsonTopic, value.get()));
+                        deserializedRawBsm.setValidatorResults(validationResults);
+                        logger.debug(validationResults.describeResults());
+                    } catch (Exception e) {
+                        JsonValidatorResult validatorResult = new JsonValidatorResult();
+
+                        validatorResult.addException(e);
+                        deserializedRawBsm.setValidationFailure(true);
+                        deserializedRawBsm.setValidatorResults(validatorResult);
+                        deserializedRawBsm.setFailedMessage(e.getMessage());
+
+                        logger.error("Error in bsmValidation:", e);
+                    }
+                    return deserializedRawBsm;
+                }
+            );
+
+        // Convert ODE BSM to GeoJSON
+        KStream<RsuLogKey, ProcessedBsm<Point>> processedJsonBsmStream =
+            validatedOdeBsmStream.transform(
+                () -> new BsmProcessedJsonConverter()
+            );
+
+            processedJsonBsmStream.to(
+            // Push the joined GeoJSON stream back out to the BSM GeoJSON topic 
+            bsmProcessedJsonTopic, 
+            Produced.with(
+                JsonSerdes.RsuLogKey(),
+                JsonSerdes.ProcessedBsm())
```

Comment:
Partitioner needed here

Reply:
I was thinking it wouldn't be too effective with BSMs.

Reply:
I made one and then deleted it actually lol.

Reply:
IMO it would be beneficial to group the BSMs into partitions however possible instead of having them randomly distributed.  It makes streams processing more efficient.

Reply:
I added it back. I called it a generic partitioner because it really can only use strings to go partition from and it makes more sense to just make it generic.

Reply:
OK I see what you mean.  That generic partitioner doesn't really do anything.  I was thinking more to add something like the RsuIdPartitioner that checks if the  key implements RsuIdKey and partitions on that if present and falls back to choose another field to use, or falls back to the default if missing.

Reply:
The existing implementation of RsuIdPartitioner would fail if that interface were implemented but RsuId were null, that would need to be fixed (with null check) since that's a possibility..

Reply:
@iyourshaw Okay, I didn't notice the RsuIdPartitioner originally. I used the same methodology with coreData id value (bsmId) as the partition decider. This should work regardless of the presence of a rsuId or a logId.